tests/sfko/sfko/obj/object.py

```diff
@@ -0,0 +1,167 @@
+import hashlib
+import uuid
+from binascii import unhexlify
+
+from ..util.conf import config
+from ..util.parse import parse_filesize
+
+
+class FakeFile:
+    def __init__(self, size, content = b'0', parent=None):
+        self._size = size
+        self._pos = 0
+        self._content = bytes(content)
+        self._parent = parent
+
+    def _etag(self, data):
+        # print('-'*4)
+        if self._parent:
+            etag = hashlib.md5(data).hexdigest()
+            # print(etag)
+            self._parent._report_etag(etag)
+        return data
+
+    def _makebytes(self, num):
+        self._pos += num
+        return self._content * num
+
+    def _readbytes(self, num):
+        return self._etag(self._makebytes(num))
+
+    def read(self, num = -1):
+        if num == -1:
+            return self._readbytes(self._size)
+        if self._pos == self._size:
+            return self._readbytes(0)
+        elif self._size < self._pos + num:
+            return self._readbytes(self._size - self._pos)
+        return self._readbytes(num)
+
+    def seek(self, offset, from_what = 0):
```

Comment:
I think there are python constants in the `os` module that could be used for the `from_what` value (like `SEEK_SET` or `SEEK_CUR`)